surah/src/commonMain/kotlin/com/islamversity/surah/SurahProcessor.kt

```diff
@@ -23,17 +21,42 @@ class SurahProcessor(
 
     override fun transformers(): List<FlowBlock<SurahIntent, SurahResult>> = listOf(
         getMainAyaProcessor,
+        bismillahTypeProcessor,
     )
 
     private val getMainAyaProcessor: FlowBlock<SurahIntent, SurahResult> = {
         ofType<SurahIntent.Initial>()
             .flatMapMerge {
                 getAyaUseCase.observeAyaMain(SurahID(it.surahId))
                     .mapListWith(ayaMapper)
-                    .map { ayas ->
+                    .map {
                         val fontSize = settingRepo.getCurrentFontSize()
-                        SurahResult.MainAyasLoaded(ayas, it.startingAyaPosition, fontSize.size.toInt())
+                        it.map {
+                            it.copy(fontSize = fontSize.size.toInt())
+                        } to fontSize
+                    }
+                    .map { ayas ->
+                        SurahResult.MainAyasLoaded(ayas.first, it.startingAyaPosition, ayas.second.size.toInt())
                     }
             }
     }
+
+    private val bismillahTypeProcessor: FlowBlock<SurahIntent, SurahResult> = {
+        ofType<SurahIntent.Initial>()
+            .publish({
+                map {
+                    when (it.bismillahType) {
+                        BismillahType.NEEDED -> SurahResult.Bismillah.Show
+                        BismillahType.FIRST_AYA,
+                        BismillahType.NONE -> SurahResult.Bismillah.Hide
+                    }
+                }
+            }, {
+                TODO("process bismillah and get from the database if needed")
```

Comment:
seems backstage is ready

Reply:
will be next PR
to avoid making this lengthy